platform/android/export/export_plugin.cpp

```diff
@@ -1125,7 +1125,11 @@ void EditorExportPlatformAndroid::_fix_manifest(const Ref<EditorExportPreset> &p
 					}
 
 					if (tname == "application" && attrname == "appCategory") {
-						encode_uint32(_get_app_category_value(app_category), &p_manifest.write[iofs + 16]);
+						if (app_category != APP_CATEGORY_OTHER) {
+							encode_uint32(_get_app_category_value(app_category), &p_manifest.write[iofs + 16]);
+						} else {
+							encode_uint32(-1, &p_manifest.write[iofs + 16]);
+						}
```

Comment:
Per the [documentation](https://developer.android.com/reference/android/content/pm/ApplicationInfo?hl=en#CATEGORY_UNDEFINED), `-1` is the right value, so this should work.

Let's update `_get_app_category_value` instead and add a case for `CATEGORY_UNDEFINED`

Reply:
I knew -1 is given for undefined category but couldn't find the documentation for it. I saw [this](https://developer.android.com/guide/topics/manifest/application-element#appCategory) and thought android team removed it or something 😅

Reply:
@m4gr3d I have updated this PR with the requested changes. Let me know if there's anything else you'd like me to tweak.

Note: I've intentionally retained the logic for Gradle builds, as I believe omitting the appCategory in the manifest is a better approach.